src/overlays/actors/ovl_En_Fg/z_en_fg.h

```diff
@@ -5,11 +5,59 @@
 
 struct EnFg;
 
+typedef void (*EnFgActionFunc)(struct EnFg*, GlobalContext*);
+
+// Based on the envColor data. Related to params but mostly unused.
+typedef enum {
+    /* 0x00 */ FG_YELLOW,
+    /* 0x01 */ FG_CYAN,
+    /* 0x02 */ FG_PINK,
+    /* 0x03 */ FG_BLUE,
+    /* 0x04 */ FG_WHITE,
+    /* 0x05 */ FG_BLACK, // All frogs are blackened when hit by an explosion
+} FrogType;
+
+typedef enum {
+    /* 0x00 */ FG_DMGEFFECT_NONE,
+    /* 0x01 */ FG_DMGEFFECT_EXPLOSION, // Bomb or bombchu, not powderkeg
+    /* 0x02 */ FG_DMGEFFECT_DEKUSTICK,
+    /* 0x03 */ FG_DMGEFFECT_HOOKSHOT,
+    /* 0x04 */ FG_DMGEFFECT_ARROW,
+    /* 0x05 */ FG_DMGEFFECT_ICEARROW,
+} FrogDamageEffect;
+
+// TODO: This is a common struct and may be moved to a more global header
+// Typically the frame/transition data is f32, not s16. 
+typedef struct {
+    /* 0x00 */ AnimationHeader* animationSeg;
+    /* 0x04 */ f32 playbackSpeed;
+    /* 0x08 */ s16 frame;
+    /* 0x0A */ s16 frameCount;
+    /* 0x0C */ u8 mode;
+    /* 0x0E */ s16 transitionRate;
+} EnFgAnimation; // size = 0x10
+
+typedef struct {
+    /* 0x0000 */ u8 type;
+    /* 0x0001 */ u8 timer;
+    /* 0x0004 */ f32 xyScale;
+    /* 0x0008 */ char unk_08[0xC];
+    /* 0x0014 */ Vec3f pos;
+    /* 0x0020 */ Vec3f unk_20; // Likely acceleration, set to 0 but unused
+    /* 0x002C */ Vec3f velocity;
+    /* 0x0038 */ char unk_38[0x4];
+} EnFgEffectDust; // size = 0x3C
+
 typedef struct EnFg {
     /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x410];
+    /* 0x144 */ EnFgActionFunc actionFunc;
+    /* 0x148 */ SkelAnime skelAnime;
+    /* 0x18C */ ColliderCylinder collider;
+    /* 0x1D8 */ Vec3s limbDrawTbl[24];
+    /* 0x268 */ Vec3s transitionDrawTbl[24];
+    /* 0x2F8 */ s16 timer;
+    /* 0x2FA */ s16 unk_2FA; // Scales y-velocity of actor
```

Comment:
`accel`

Reply:
This really isn't acceleration though, It's more like a discrete value of velocity. One would expect accel to be:
`this->actor.velocity.y  += accel;`
when in actuality we have:
`this->actor.velocity.y = 10.0f / this->unk_2FA;'
where unk_2FA is a discrete number that cycles through 1-4

Reply:
I think this is how many times the frog has hit the ground after being exploded. Since it is being incremented when `(this->actor.velocity.y <= 0.0f) && (this->actor.bgCheckFlags & 1)` is true, which looks like the exact same check when the frog is falling to go back to waiting to jump.

This would also explain the 10.0f / this->unk_2FA as the frog is bouncing up again but slower then previously.

Reply:
Again poor frogs.

Reply:
Ah @hensldm it does indeed look like a bounce count! Good catch.